tests/test_requests.py

```diff
@@ -1350,6 +1350,16 @@ def test_cookie_duplicate_names_raises_cookie_conflict_error(self):
         with pytest.raises(requests.cookies.CookieConflictError):
             jar.get(key)
 
+    def test_cookies_popitem(self):
+        jar = requests.cookies.RequestsCookieJar()
+        with pytest.raises(KeyError):
+            jar.popitem()
+
+        jar.set("1st_key", "1st_value")
+        jar.set("2nd_key", "2nd_value")
+        cookie = next(iter(jar))
+        assert jar.popitem() == (cookie, cookie.value)
```

Comment:
Can you add an assertion of what the jar looks like afterwards, I think `__delitem__` needs a fix also

Reply:
Good call @graingert , `__delitem__` had the same problem.

Reply:
It looks like popitem will delete multiple items instead of just one, can you add a test for that?

Reply:
This one has been stuck for a long time. Do you have any suggestions for the `remove_cookie_by_name` problem?

Reply:
You can implement popitem directly instead of editing __delitem__ and __getitem__